src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -270,7 +274,9 @@ static void AddDiagnosticsToResult(
             }
 
             diagnostics = diagnostics.Filter(diagnosticIdsToFilter, includeSuppressedDiagnostics, span);
-            Debug.Assert(diagnostics.Length == CompilationWithAnalyzers.GetEffectiveDiagnostics(diagnostics, compilation).Count());
+
+            // ⚠️ No obvious way to keep this assertion
+            //Debug.Assert(diagnostics.Length == CompilationWithAnalyzers.GetEffectiveDiagnostics(diagnostics, compilation).Count());
```

Comment:
📝 Need to understand how important this assertion is. If it is not important, there are multiple parameters that can be removed to simplify this code.

Reply:
➡️ I removed this assertion and the parameters supporting it.